src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -217,7 +217,11 @@ public static FormulaValue TimeValue_UO(IRContext irContext, UntypedObjectValue[
             if (impl.Type == FormulaType.String)
             {
                 var s = impl.GetString();
-                if (TimeSpan.TryParseExact(s, @"hh\:mm\:ss\.FFF", CultureInfo.InvariantCulture, TimeSpanStyles.None, out TimeSpan res))
+
+                TimeSpan res;
+
+                if (TimeSpan.TryParseExact(s, @"hh\:mm\:ss\.FFFFFFF", CultureInfo.InvariantCulture, TimeSpanStyles.None, out res) ||
```

Comment:
Is this logic shared with DateTimeValue(string)? (And should it be?)

Reply:
Good question. DateTimeValue and TimeValue functions take locale as arguments. On the other hand, the untyped versions don't and always expect an invariant. That makes me think we don't need to share this logic with DateTimeValue /TimveValue.